src/gateway/router/routes/searchRoute.ts

```diff
@@ -31,10 +31,20 @@ export async function searchRoute(ctx: Router.RouterContext) {
           SELECT 4 as sort_order, src_tx_id as id, 'source' as type, '{}'::jsonb as interaction, '' as confirmation_status, '' as pst_ticker, '' as pst_name
           FROM contracts_src
           WHERE src_tx_id ILIKE ?
+          UNION ALL
+          SELECT 5 as sort_order, creator_id as id, 'creator' as type, '{}'::jsonb as interaction, '' as confirmation_status, '' as pst_ticker, '' as pst_name
+          FROM
+          (
+            WITH temp_creator AS (
+                SELECT DISTINCT owner as creator_id from interactions where owner ILIKE ?
+                UNION ALL
+                SELECT DISTINCT owner as creator_id FROM contracts WHERE owner ILIKE ?
+            )
+            SELECT DISTINCT * FROM temp_creator) AS creator
```

Comment:
not satisfied with the query time, it takes a while to go through all the interactions :(

Reply:
try adding a gin index on the new 'owner' column
`CREATE INDEX idx_interaction_owner_gin ON interactions USING gin (owner gin_trgm_ops);`

Reply:
and maybe also add it on the contracts table...